latch/registry/table.py

```diff
@@ -0,0 +1,275 @@
+import asyncio
+from dataclasses import dataclass
+from typing import Any, Dict, Iterator, List, Optional, Type, TypedDict
+
+import gql
+
+from latch.gql.execute import execute, get_transport
+from latch.registry.record import Record
+from latch.registry.utils import (
+    clean,
+    to_python_literal,
+    to_python_type,
+    to_registry_literal,
+)
+
+
+class ListRecordsOutputType(TypedDict):
+    records: List[Record]
+    errors: List[Exception]
+
+
+class Table:
+    def __init__(self, id: str):
+        self.id = id
+        self._name: Optional[str] = None
+        self._record_type: Type[Record] = None
+        self._columns: Optional[List[Dict[str, Dict]]] = None
+        self._info: Optional[Dict[str, Dict]] = None
+
+    def load(self):
+        self.get_display_name(load_if_missing=True)
+        self.get_columns(load_if_missing=True)
+
+    def _get_info(self):
+        if self._info is not None:
+            return self._info
+
+        self._info = execute(
+            """
+            query tableQuery ($argTableId: BigInt!) {
+                catalogExperiment(id: $argTableId) {
+                    id
+                    displayName
+                    catalogExperimentColumnDefinitionsByExperimentId {
+                        nodes {
+                            key
+                            type
+                        }
+                    }
+                    catalogSamplesByExperimentId(condition: { removed: false }) {
```

Comment:
don't always fetch all samples since it makes the paged version completely useless